BaseDockerfile

```diff
@@ -0,0 +1,75 @@
+FROM nvidia/cuda:11.6.0-devel-ubuntu20.04
+
+# Copied out of anaconda's dockerfile
+ARG MINICONDA_VERSION=4.6.14
+ARG MAKE_VERSION=4.2.1-1.2
+ARG GIT_VERSION=1:2.25.1-1ubuntu3.2
+ARG WGET_VERSION=1.20.3-1ubuntu2
+RUN apt-get update && apt-get install --no-install-recommends -y wget=${WGET_VERSION} git=${GIT_VERSION} make=${MAKE_VERSION} \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-${MINICONDA_VERSION}-Linux-x86_64.sh -O ~/miniconda.sh && \
+    /bin/bash ~/miniconda.sh -b -p /opt/conda && \
+    rm ~/miniconda.sh && \
+    ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh && \
+    echo ". /opt/conda/etc/profile.d/conda.sh" >> ~/.bashrc
+
+# Setup CMake
+ARG CMAKE_VERSION=3.22.1
+RUN wget --quiet https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-linux-x86_64.tar.gz -O cmake.tar.gz && \
+    tar -xzf cmake.tar.gz && \
+    rm -rf cmake.tar.gz
+
+ENV PATH $PATH:$PWD/cmake-${CMAKE_VERSION}-linux-x86_64/bin/
+
+# Copy the environment yaml to cache environment when possible
+COPY environment.yml /code/timemachine/
+
+WORKDIR /code/timemachine/
+
+ARG ENV_NAME=timemachine
+
+# Create Timemachine Env
+RUN . /opt/conda/etc/profile.d/conda.sh && \
+    conda env create -n "${ENV_NAME}" --force -f environment.yml && \
+    conda activate ${ENV_NAME}
+ENV PATH /opt/conda/envs/${ENV_NAME}/bin:$PATH
+
+ENV CONDA_DEFAULT_ENV ${ENV_NAME}
+
+# Install OpenMM
+ARG OPENMM_VERSION=7.5.1
+
+ARG DOXYGEN_VERSION=1.9.1
+ARG CYTHON_VERSION=0.29.26
+ARG SWIG_VERSION=3.0.12
+
+RUN . /opt/conda/etc/profile.d/conda.sh && conda install -y -c conda-forge swig=${SWIG_VERSION} doxygen=${DOXYGEN_VERSION} cython=${CYTHON_VERSION}
+
+WORKDIR /code/
+RUN git clone https://github.com/openmm/openmm.git --branch "${OPENMM_VERSION}" && \
+    cd openmm/ && \
+    mkdir build && \
+    cd build && \
+    cmake \
+      -DOPENMM_BUILD_CPU_LIB=OFF \
+      -DOPENMM_BUILD_AMOEBA_CUDA_LIB=OFF \
+      -DOPENMM_BUILD_AMOEBA_OPENCL_LIB=OFF \
+      -DOPENMM_BUILD_CUDA_LIB=OFF \
+      -DOPENMM_BUILD_CUDA_COMPILER_PLUGIN=OFF \
+      -DOPENMM_BUILD_C_AND_FORTRAN_WRAPPERS=OFF \
+      -DOPENMM_BUILD_DRUDE_CUDA_LIB=OFF \
+      -DOPENMM_BUILD_DRUDE_OPENCL_LIB=OFF \
+      -DOPENMM_BUILD_EXAMPLES=OFF \
+      -DOPENMM_BUILD_OPENCL_LIB=OFF \
+      -DOPENMM_BUILD_RPMD_CUDA_LIB=OFF \
+      -DOPENMM_BUILD_RPMD_OPENCL_LIB=OFF \
+      -DCMAKE_INSTALL_PREFIX=../openmm_install \
+      ../ && \
+    make -j "$(nproc)" && \
+    make -j "$(nproc)" install && \
```

Comment:
Is it possible to move the OpenMM build before the creation of the timemachine conda env? (would probably still need to have the base env activated..) The hope would be to avoid needing to rebuild OpenMM from scratch if e.g. there's only a change to `environment.yml`

Reply:
Tangentially, I wonder if there's a way to move the OpenMM stuff into a separate Dockerfile using a [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds)? I haven't done this before, but it seems potentially nice for modularization (maybe at the cost of having more images to track)

(not intended to be a PR blocker)